hikari/impl/shard.py

```diff
@@ -53,14 +46,246 @@
+_DISPATCH: typing.Final[int] = 0
+_HEARTBEAT: typing.Final[int] = 1
+_IDENTIFY: typing.Final[int] = 2
+_PRESENCE_UPDATE: typing.Final[int] = 3
+_VOICE_STATE_UPDATE: typing.Final[int] = 4
+_RESUME: typing.Final[int] = 6
+_RECONNECT: typing.Final[int] = 7
+_REQUEST_GUILD_MEMBERS: typing.Final[int] = 8
+_INVALID_SESSION: typing.Final[int] = 9
+_HELLO: typing.Final[int] = 10
+_HEARTBEAT_ACK: typing.Final[int] = 11
+# If we disconnect within this period of time after starting, we should
+# use an exponential backoff before restarting.
+_BACKOFF_WINDOW: typing.Final[float] = 30.0
+_BACKOFF_BASE: typing.Final[float] = 1.85
+_BACKOFF_INCREMENT_START: typing.Final[int] = 2
+_BACKOFF_CAP: typing.Final[float] = 600.0
+# Ping-pong. Might disable this eventually.
+_PING_PONG_HEARTBEAT_LATENCY: typing.Optional[float] = 20.0
+# Discord seems to invalidate sessions if I send a 1xxx, which is useless
+# for invalid session and reconnect messages where I want to be able to
+# resume.
+_RESUME_CLOSE_CODE: typing.Final[int] = 3_000
+# Per-shard sending rate-limit
+_TOTAL_RATELIMIT: typing.Final[typing.Tuple[float, int]] = (60.0, 120)
+# Rate-limit for chunking requests (used to prevent saturating the entire
+# ratelimit window).
+_CHUNKING_RATELIMIT: typing.Final[typing.Tuple[float, int]] = (60.0, 60)
+
+
+@typing.final
+class _V6GatewayTransport(aiohttp.ClientWebSocketResponse):
+    """Internal component to handle lower-level communication logic.
+
+    This includes translating aiohttp error conditions to hikari ones,
+    handling inbound zlib packets, creating the websocket and client session,
+    and ensuring all resources are freed deterministically where possible.
+
+    Payload logging is also performed here.
+    """
+
+    # Initialized from `connect'
+    _zlib: zlib._Decompress
+    _logger: logging.Logger
+    _debug: bool
+
+    def __init__(self, *args: typing.Any, **kwargs: typing.Any) -> None:
+        super().__init__(*args, **kwargs)
+        self._zlib = zlib.decompressobj()
+
+    def close(self, *, code: int = 1000, message: bytes = b"") -> typing.Coroutine[typing.Any, typing.Any, bool]:
+        if not self._closed and not self._closing:
+            self._logger.debug("sending close frame with code %s and message %s", code, message)
+        return super().close(code=code, message=message)
+
+    async def receive_json(
+        self,
+        *,
+        loads: aiohttp.typedefs.JSONDecoder = json.loads,
+        timeout: typing.Optional[float] = None,
+    ) -> typing.Any:
+        pl = await self._receive_and_check(timeout)
+        self._log_debug_payload(pl, "received")
+        return loads(pl)
+
+    async def send_json(
+        self,
+        data: data_binding.JSONObject,
+        compress: typing.Optional[int] = None,
+        *,
+        dumps: aiohttp.typedefs.JSONEncoder = json.dumps,
+    ) -> None:
+        pl = dumps(data)
+        self._log_debug_payload(pl, "sending")
+        await super().send_str(pl, compress)
```

Comment:
does this have to be a super call?

Reply:
No, I have already fixed this in my testing branch :+1: